deals/module/deals.go

```diff
@@ -583,10 +590,10 @@ func (m *Module) eventuallyFinalizeDeal(dr deals.StorageDealRecord, timeout time
 					log.Errorf("storing proposal cid %s deal record: %v", util.CidToString(info.ProposalCid), err)
 				}
 				return
-			} else if info.StateID == storagemarket.StorageDealProposalNotFound ||
-				info.StateID == storagemarket.StorageDealProposalRejected ||
-				info.StateID == storagemarket.StorageDealFailing {
-				log.Infof("proposal cid %s failed with state %s, deleting pending deal", util.CidToString(info.ProposalCid), storagemarket.DealStates[info.StateID])
+			} else if info.Status == storagemarket.StorageDealProposalNotFound ||
+				info.Status == storagemarket.StorageDealProposalRejected ||
+				info.Status == storagemarket.StorageDealFailing {
+				log.Infof("proposal cid %s failed with state %s, deleting pending deal", util.CidToString(info.ProposalCid), storagemarket.DealStates[info.Status])
```

Comment:
Not important for the feedback I'm trying to gather now. Long story short, was experimenting with using the `storagemarket` deal state enums throughout the stack since they will be a part of data exposed by the API. Not sure if using that enum directly is good or bad, or we could mirror it with our own enum type.

Reply:
Might be trickier whenever we support other Filecoin clients, but maybe we can always keep the mapping.

Reply:
Agree it could be tricky, but we're already in the same situation because we're simply using the raw enum value and string representations, all provided by `storeagemarket`... We don't have an enum of our own. We probably should.